From c4a78dfa1cacb4a860f338e552b7b8f4d5a17edb Mon Sep 17 00:00:00 2001
Message-Id: <c4a78dfa1cacb4a860f338e552b7b8f4d5a17edb.1424979244.git.feitong.yi@intel.com>
In-Reply-To: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
References: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Arun Siluvery <arun.siluvery@linux.intel.com>
Date: Thu, 19 Feb 2015 11:08:57 +0000
Subject: [PATCH 25/28] SQUASHME! [VPG]: drm/i915: Keep WA init functions
 global as in original patch

The WA init functions were initially exported but when this patch
is reverted to integrate upstream patches they were made static
as they are used only in ring buffer mode. A patch that was supposed
to revert also ammended the patch which creates a discrepancy when
folding those reverts. This patch removes the ammended changes to
complete the revert properly.

This patch should be squashed into the following patch,

     Author: Arun Siluvery <arun.siluvery@linux.intel.com>
     Date:   Thu Nov 13 17:58:42 2014 +0000
     Change-Id: I93770fc0f91fa979dfd7ff2a291060cd32cb7284
     Revert "FOR_UPSTREAM [VPG]: drm/i915: Initialize bdw workarounds in logical ring mode too"

For: VIZ-4092
Change-Id: I7a661bc21b7a80139a87232848971045a1bcb8c1
Signed-off-by: Arun Siluvery <arun.siluvery@linux.intel.com>
---
 drivers/gpu/drm/i915/intel_ringbuffer.c |    4 ++--
 drivers/gpu/drm/i915/intel_ringbuffer.h |    2 ++
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_ringbuffer.c b/drivers/gpu/drm/i915/intel_ringbuffer.c
index 9a98990..1d9af73 100644
--- a/drivers/gpu/drm/i915/intel_ringbuffer.c
+++ b/drivers/gpu/drm/i915/intel_ringbuffer.c
@@ -1228,7 +1228,7 @@ static int wa_add(struct drm_i915_private *dev_priv,
 
 #define WA_WRITE(addr, val) WA_REG(addr, val, 0xffffffff)
 
-static int bdw_init_workarounds(struct intel_engine_cs *ring)
+int bdw_init_workarounds(struct intel_engine_cs *ring)
 {
 	struct drm_device *dev = ring->dev;
 	struct drm_i915_private *dev_priv = dev->dev_private;
@@ -1273,7 +1273,7 @@ static int bdw_init_workarounds(struct intel_engine_cs *ring)
 	return 0;
 }
 
-static int chv_init_workarounds(struct intel_engine_cs *ring)
+int chv_init_workarounds(struct intel_engine_cs *ring)
 {
 	struct drm_device *dev = ring->dev;
 	struct drm_i915_private *dev_priv = dev->dev_private;
diff --git a/drivers/gpu/drm/i915/intel_ringbuffer.h b/drivers/gpu/drm/i915/intel_ringbuffer.h
index 368bdea..3ff4450 100644
--- a/drivers/gpu/drm/i915/intel_ringbuffer.h
+++ b/drivers/gpu/drm/i915/intel_ringbuffer.h
@@ -647,6 +647,8 @@ int intel_init_vebox_ring_buffer(struct drm_device *dev);
 u64 intel_ring_get_active_head(struct intel_engine_cs *ring);
 void intel_ring_setup_status_page(struct intel_engine_cs *ring);
 u32 get_pipe_control_scratch_addr(struct intel_engine_cs *ring);
+int bdw_init_workarounds(struct intel_engine_cs *ring);
+int chv_init_workarounds(struct intel_engine_cs *ring);
 int init_workarounds_ring(struct intel_engine_cs *ring);
 
 
-- 
1.7.9.5

